addons/mass_mailing/models/mass_mailing.py

```diff
@@ -746,34 +578,34 @@ def get_recipients(self, cr, uid, mailing, context=None):
             res_ids = random.sample(remaining, topick)
         return res_ids
 
-    def get_remaining_recipients(self, cr, uid, mailing, context=None):
-        res_ids = self.get_recipients(cr, uid, mailing, context=context)
-        already_mailed = self.pool['mail.mail.statistics'].search_read(cr, uid, [('model', '=', mailing.mailing_model),
-                                                                                 ('res_id', 'in', res_ids),
-                                                                                 ('mass_mailing_id', '=', mailing.id)], ['res_id'], context=context)
+    def get_remaining_recipients(self):
+        res_ids = self.get_recipients()
+        already_mailed = self.env['mail.mail.statistics'].search_read([('model', '=', self.mailing_model),
+                                                                     ('res_id', 'in', res_ids),
+                                                                     ('mass_mailing_id', '=', self.id)], ['res_id'])
         already_mailed_res_ids = [record['res_id'] for record in already_mailed]
         return list(set(res_ids) - set(already_mailed_res_ids))
 
-    def send_mail(self, cr, uid, ids, context=None):
-        author_id = self.pool['res.users'].browse(cr, uid, uid, context=context).partner_id.id
-        for mailing in self.browse(cr, uid, ids, context=context):
+    def send_mail(self):
+        author_id = self.env.user.partner_id.id
+        for mailing in self:
             # instantiate an email composer + send emails
-            res_ids = self.get_remaining_recipients(cr, uid, mailing, context=context)
+            res_ids = mailing.get_remaining_recipients()
             if not res_ids:
                 raise UserError(_('Please select recipients.'))
 
-            if context:
-                comp_ctx = dict(context, active_ids=res_ids)
+            if self.env.context:
+                comp_ctx = dict(self.env.context, active_ids=res_ids)
             else:
                 comp_ctx = {'active_ids': res_ids}
 
             # Convert links in absolute URLs before the application of the shortener
-            self.write(cr, uid, [mailing.id], {'body_html': self.pool['mail.template']._replace_local_links(cr, uid, mailing.body_html, context)}, context=context)
+            mailing.body_html = self.env['mail.template']._replace_local_links(mailing.body_html)
 
             composer_values = {
                 'author_id': author_id,
                 'attachment_ids': [(4, attachment.id) for attachment in mailing.attachment_ids],
-                'body': self.convert_links(cr, uid, [mailing.id], context=context)[mailing.id],
+                'body': self.convert_links()[mailing.id],
```

Comment:
I think it should be mail.convert_links()[mailing.id]. I'm pretty sure it will crash in multi record